server/repository/src/db_diesel/item_row.rs

```diff
@@ -85,6 +97,8 @@ impl<'a> ItemRowRepository<'a> {
             .do_update()
             .set(item_row)
             .execute(&self.connection.connection)?;
+
+        insert_or_ignore_item_link(&self.connection, item_row)?;
```

Comment:
We could possibly remove `insert_or_ignore_item_link` call a one-liner instead pretty cleanly if we implemented  `ItemLinkRow::from(ItemRow)` trait. I did go down this road but I didn't know what the expectation is around when to use references:

```rs
// item_link_row.rs
impl From<ItemRow> for ItemLinkRow {
    fn from(item: ItemRow) -> Self {
        Self {
            id: item.id.clone(),
            item_id: item.id.clone(),
        }
    }
}

// Here
ItemLinkRowRepository::new(&self.connection)
            .insert_one_or_ignore(&ItemLinkRow::from(item_row.clone()))?;
```
Sooo much cloning though? Kinda ugly? OR:

```rs
// item_link_row.rs
impl From<&ItemRow> for ItemLinkRow {
    fn from(item: &ItemRow) -> Self {
        Self {
            id: item.id.clone(),
            item_id: item.id.clone(),
        }
    }
}

// Here
ItemLinkRowRepository::new(&self.connection)
            .insert_one_or_ignore(&ItemLinkRow::from(item_row))?;
```

So naturally passing a reference has less cloning, but is it normal to implement `From<&T>` or `From<T>`?

Reply:
I suppose it wouldn't be incorrect to do both!

Reply:
I think cloning should be fine here, and it's pretty clear ot me, without clone that there is a conversion from item row to item link row. I think from becomes usefull for common things where logic is repeated many times over, or maybe for generic things that do something to unknown data type that has to have a 'from' property (like collect() for iterator)